src/main/java/swinglib/ErrorMessageBuilder.java

```diff
@@ -0,0 +1,123 @@
+package swinglib;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import java.awt.GraphicsEnvironment;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JOptionPane;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+
+/**
+ * Swing builder for an error message pop-up (Uses JOptionPane). Example usages:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .title("Something Crashed")
+ *     .message("Error Details")
+ *     .option("ok", () -> {})
+ *     .option("quit", () -> System.exit(0))
+ *     .buildAndShow();
+ * </pre></code>
+ * Default error dialog with an 'ok' button:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .message("error details")
+ *     .buildAndShow();
+ * </pre></code>
+ */
+public class ErrorMessageBuilder {
+
+  private String title = "Error";
+  private String message;
+  private List<CloseButton> actions = new ArrayList<>();
+
+  private static class CloseButton {
+    private final String buttonText;
+    private final Runnable closeAction;
+
+    private CloseButton(String buttonText, Runnable closeAction) {
+      this.buttonText = buttonText;
+      this.closeAction = closeAction;
+    }
+  }
+
+  private ErrorMessageBuilder() {}
+
+  public static ErrorMessageBuilder builder() {
+    return new ErrorMessageBuilder();
+  }
+
+  public ErrorMessageBuilder message(String message) {
+    this.message = message;
+    return this;
+  }
+
+  /**
+   * Constructs a Swing JDialog using current builder values.
+   * Values that must be set: title, contents
+   */
+  public void buildAndShow() {
```

Comment:
Is there a guarantee someplace that this method will be invoked on the EDT?  Seems like it could be a problem if it's not given that the `ClientLogger#logXXX()` methods might be called from multiple threads.

Reply:
No guarantee as this is effectively library code and could be used about anywhere we do swing component creation. Avoiding that problem is a good detail to take care, reduces the chance of error when using this api. In 41d5147c6 I took the suggestion and wrapped the `buildAndShow` with a `SwingAction.invokeAndWait` call.